packages/blade/src/components/Toast/_decisions/decisions.md

```diff
@@ -0,0 +1,267 @@
+
+```ts
+type useToastReturnType = {
+  /**
+   * @returns id of the toast
+   */
+  show: (toast: Toast) => string;
+
+  /**
+   * id of the toast to be dismissed
+   * 
+   * if id is not provided, all the toasts will be dismissed
+   */
+  dismiss: (toastId?: string) => void;
+}
+```
+
+## Examples
+
+#### Removing a toast
+
+react-hot-toast provides this functionality, for more info see [react-hot-toast docs](https://react-hot-toast.com/docs/toast#dismiss-toast-programmatically)
+
+```jsx
+import { BladeProvider, ToastContainer, useToast } from "@razorpay/blade/components"
+
+const Example = () => {
+  const toastId = React.useRef(null);
+  const toast = useToast();
+
+  return (
+    <Box>
+      <Button
+        onClick={() => {
+          toastId.current = toast.show({
+            color: 'success',
+            content: 'Payment Successful',
+          });
+        }}
+      >
+        Show Toast
+      </Button>
+      <Button onClick={() => toast.dismiss(toastId.current)}>Dismiss Toast</Button>
+    </Box>
+  );
+};
+```
+
+### Promotional Toast
+
+```jsx
+import { BladeProvider, ToastContainer, useToast } from "@razorpay/blade/components"
+
+const Example = () => {
+  const toast = useToast();
+
+  return (
+    <Button
+      onClick={() => {
+        toast.show({
+          type: 'promotional',
+          content: (
+            <Box>
+              <Heading>Payment Successful</Heading>
+              <Text>Amount: ₹100</Text>
+            </Box>
+          ),
+          leading: <DollarIcon />,
+          action: {
+            text: 'Okay'
+          }
+        });
+      }}
+    >
+      Do payment
+    </Button>
+  );
+};
+```
+
+<img src="./example-promotional-toast.png" width="380" alt="Promotional toast example" />
+
+
+## Motion
+
+You can checkout the toast motion [here](https://www.figma.com/proto/jubmQL9Z8V7881ayUD95ps/Blade-DSL?type=design&node-id=75848-1063056&t=QZkki0ZrlcG4sKzw-0&scaling=min-zoom&page-id=7665%3A27414).
+
+## Accessibility
+
+- Toast components will follow the WAI-ARIA guidelines for [alert](https://www.w3.org/WAI/ARIA/apg/patterns/alert/examples/alert/). For error toast we will use the `alert` role and for informational toast we will use the `status` role.
+
+
+## References
+
+- https://react-hot-toast.com/docs
+- https://chakra-ui.com/docs/components/toast/usage
+
+## Open Questions
+
+- Q. What should be the default duration for auto dismissable toasts?
+  - A. 4s for informational toasts and 8s for promotional toasts
+
+- Q. Should we call it `onDismissButtonClick` or `onDismiss`? 
+  
+- Q. Should the dismiss handler be called even when the toast is auto dismissed? Or should we have different handlers for auto dismiss and manual dismiss? (eg: `onAutoDismiss` `onDismissButtonClick`)
+  - A. We will call it `onDismissButtonClick` and it will only be called when the user clicks on the dismiss button.
```

Comment:
if product needs to track the event on auto dismiss then how can consumer handle that??

Reply:
you can do multiple things like for eg: a user was shown a toast and they navigate to a different page then when the toast is dismissed you can track the URL of the page. this way you can identify a pattern that after performing some actions what are the pages a user navigates to and get some insights around user behavior

Reply:
> a user was shown a toast and they navigate to a different page then when the toast is dismissed you can track the URL of the page

That is possible. 

`onDismissButtonClick={() => naigateToPage()}`

But don't think it will be a very good idea for toast to automatically navigate the page after dismiss. Navigation should happen after action.

Reply:
FYI, 

react-hot-toast doesn't have a [callback](https://github.com/timolins/react-hot-toast/issues/174) that it calls when a toast gets auto dismissed, so it's not even possible to have an `onAutoDismiss` method. 

There is a https://github.com/timolins/react-hot-toast/issues/63#issuecomment-1344772935 but doing that means you need to keep track of each and every toast's timer manually, too messy.